core/src/avm2/optimize.rs

```diff
@@ -0,0 +1,893 @@
+                num_args,
+            } => {
+                // Arguments
+                stack.popn(*num_args);
+
+                let multiname = stack_pop_multiname!(name_index);
+
+                // Then receiver.
+                let stack_value = stack.pop();
+
+                if let Some(multiname) = multiname {
+                    if !multiname.has_lazy_component() {
+                        if let Some(ValueType::Class(class)) = stack_value {
+                            if !class.inner_class_definition().read().is_interface() {
+                                match class.instance_vtable().get_trait(&multiname) {
+                                    Some(Property::Method { disp_id }) => {
+                                        *op = Op::CallMethod {
+                                            num_args: *num_args,
+                                            index: Index::new(disp_id),
+                                        };
+                                    }
+                                    _ => {}
+                                }
+                            }
+                        }
+                    }
+                    // `stack_pop_multiname` handled lazy
+                }
+
+                // Avoid checking return value for now
+                stack.push_any();
+            }
+            Op::CallPropVoid { .. } => {
+                // Avoid handling for now
+                stack.clear();
+            }
+            Op::Call { num_args } => {
+                // Arguments
+                stack.popn(*num_args);
+
+                stack.pop();
+
+                // Avoid checking return value for now
+                stack.push_any();
+            }
+            Op::GetGlobalScope => {
+                // Avoid handling for now
+                stack.push_any();
+            }
+            Op::NewActivation => {
+                // Avoid handling for now
+                stack.push_any();
+            }
+            Op::Nop => {}
+            Op::DebugFile { .. } => {}
+            Op::DebugLine { .. } => {}
+            Op::Debug { .. } => {}
+            Op::IfTrue { .. } | Op::IfFalse { .. } => {
+                stack.pop();
+            }
+            Op::IfStrictEq { .. }
+            | Op::IfStrictNe { .. }
+            | Op::IfEq { .. }
+            | Op::IfNe { .. }
+            | Op::IfGe { .. }
+            | Op::IfGt { .. }
+            | Op::IfLe { .. }
+            | Op::IfLt { .. }
+            | Op::IfNge { .. }
+            | Op::IfNgt { .. }
+            | Op::IfNle { .. }
+            | Op::IfNlt { .. } => {
+                stack.pop();
+                stack.pop();
+            }
+            Op::Si8 => {
+                stack.pop();
+                stack.pop();
+            }
+            Op::Li8 => {
+                stack.pop();
+                stack.push_int();
+            }
+            Op::ReturnVoid | Op::ReturnValue | Op::Jump { .. } | Op::LookupSwitch(_) => {
```

Comment:
note: `throw` also should terminate the block (both here and in the future refactors)